ruby_on_rails/cucumber.md

```diff
@@ -0,0 +1,86 @@
+# Cucumber
+
+[Cucumber](https://cucumber.io) is a testing framework which allows us to specify feature tests using plain language.
+In our projects, we use Cucumber for e2e tests. We integrate it using the
+`cucumber-rails` gem.
+
+## Installing `cucumber-rails`
+
+The installation process of `cucumber-rails` is also documented on its
+[Github page](https://github.com/cucumber/cucumber-rails).
+
+To install `cucumber-rails`, add the following gems to your Gemfile:
+
+```rb
+group :test do
+    gem 'capybara', require: false
+    gem 'cucumber-rails', require: false
+    gem 'database-cleaner'
+end
+```
+
+Note that `capybara` and `database-cleaner` are optional. Most probably, you are going to need them.
+
+Run `bundle install` to install the gems.
+
+Finally, set up Cucumber with
+
+```sh
+rails generate cucumber:install
+```
+
+At this point, it might be a good idea to inspect the generated files and commit. At least, ckeck if you run into any
+rubocop issues. Probably, you have to add an exception in `.rubocop.yml`:
+
+```yaml
+AllCops:
```

Comment:
too generic for me, also stating "probably" is not clear. Why is it causing rubocop issues? and which ones?

Reply:
The file `.rubocop.yml` which is generated by `cucumber-rails` has lines which are too long. What do you think about rephrasing it as follows?

> At this time, it is recommended to inspect the generated files and commit. Before committing, make sure, Rubocop does not raise any issues. Therefore, you will have to add the following exception to `.rubocop.yml`:
> …

Reply:
better. yes. I would prefer to fix the issues rather than adding an exception to rubocop. Which file is causing the issue? You wrote `.rubocop.yml` but I think you meant something different 😄

Reply:
`cucumber.rake` of course. I would not bother fixing the line length in this case, as the file is generated automatically by `cucumber-rails`.

Reply:
mmmm. I see your point but I disagree because we also fix rubocop issues from the files generated by Rails (for example). 
Apart from that if the line length is the only issue I'd rather propose to disable that specific cop (line length) for that file rather than `AllCops`

Reply:
I had another look, there are two cops failing: `Metrics/LineLength` and `Rails/FilePath`

Furthermore, he file in question starts with the following comment:

```
# IMPORTANT: This file is generated by cucumber-rails - edit at your own peril.
# It is recommended to regenerate this file in the future when you upgrade to a
# newer version of cucumber-rails. Consider adding your own code to a new file
# instead of editing this one. Cucumber will automatically load all features/**/*.rb
# files.
```

I would thus propose to treat it like an external library. This means, we don't write tests for it and we don't apply our coding guidelines. If we think there is something wrong with the file, we should open a PR on `cucumber-rails`.